From: Kai Uwe Broulik <kde@privat.broulik.de>
Date: Sat, 14 Nov 2015 16:24:38 +0000
Subject: Show fully charged message when battery state becomes "Fully charged" rather than "Not charging"
X-Git-Tag: v5.4.95
X-Git-Url: http://quickgit.kde.org/?p=powerdevil.git&a=commitdiff&h=4c66646761834236f6b14789162fe71f6179f472
---
Show fully charged message when battery state becomes "Fully charged" rather than "Not charging"

BUG: 354623
FIXED-IN: 5.5.0
---


--- a/daemon/powerdevilcore.cpp
+++ b/daemon/powerdevilcore.cpp
@@ -415,11 +415,7 @@
     qCDebug(POWERDEVIL) << "A new battery was detected";
 
     m_batteriesPercent[udi] = b->chargePercent();
-    if (b->chargeState() == Solid::Battery::NoCharge) {
-      m_batteriesCharged[udi] = true;
-    } else {
-      m_batteriesCharged[udi] = false;
-    }
+    m_batteriesCharged[udi] = (b->chargeState() == Solid::Battery::FullyCharged);
     m_loadedBatteriesUdi.append(udi);
 
     const int chargePercent = currentChargePercent();
@@ -581,21 +577,21 @@
 void Core::onBatteryChargeStateChanged(int state, const QString &udi)
 {
     bool previousCharged = true;
-    for (QHash<QString,bool>::const_iterator i = m_batteriesCharged.constBegin(); i != m_batteriesCharged.constEnd(); ++i) {
+    for (auto i = m_batteriesCharged.constBegin(); i != m_batteriesCharged.constEnd(); ++i) {
         if (!i.value()) {
             previousCharged = false;
             break;
         }
     }
 
-    m_batteriesCharged[udi] = (state == Solid::Battery::NoCharge);
+    m_batteriesCharged[udi] = (state == Solid::Battery::FullyCharged);
 
     if (m_backend->acAdapterState() != BackendInterface::Plugged) {
         return;
     }
 
     bool currentCharged = true;
-    for (QHash<QString,bool>::const_iterator i = m_batteriesCharged.constBegin(); i != m_batteriesCharged.constEnd(); ++i) {
+    for (auto i = m_batteriesCharged.constBegin(); i != m_batteriesCharged.constEnd(); ++i) {
         if (!i.value()) {
             currentCharged = false;
             break;

